examples/integrations/cdi/datasource-hikaricp-mysql/pom.xml

```diff
@@ -0,0 +1,213 @@
+                                    </includes>
+                                </resource>
+                                <resource>
+                                    <filtering>true</filtering>
+                                    <directory>src/main/k8s</directory>
+                                    <includes>
+                                        <include>datasource.secrets.template</include>
+                                        <include>app.yaml</include>
+                                    </includes>
+                                </resource>
+                            </resources>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <systemPropertyVariables>
+                    </systemPropertyVariables>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
+        <!-- Compile-scoped dependencies. -->
+        <dependency>
+            <groupId>javax.enterprise</groupId>
+            <artifactId>cdi-api</artifactId>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>javax.ws.rs</groupId>
+            <artifactId>javax.ws.rs-api</artifactId>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>redis.clients</groupId>
+            <artifactId>jedis</artifactId>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.microprofile.config</groupId>
+            <artifactId>microprofile-config-api</artifactId>
+            <scope>compile</scope>
+        </dependency>
+
+        <!-- Runtime-scoped dependencies. -->
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <scope>runtime</scope>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.weld.se</groupId>
+            <artifactId>weld-se-core</artifactId>
+            <scope>runtime</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.jboss.spec.javax.el</groupId>
+                    <artifactId>jboss-el-api_3.0_spec</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.jboss.spec.javax.interceptor</groupId>
+                    <artifactId>jboss-interceptors-api_1.2_spec</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.helidon.integrations.cdi</groupId>
+            <artifactId>helidon-integrations-cdi-datasource-hikaricp</artifactId>
+            <version>${project.version}</version>
```

Comment:
Our examples are inconsistent on how we handle versioning for the Helidon dependencies. Some use `project.version`, some define and use `helidon.version` (see webserver examples) and others use `helidon.version` in conjunction with dependency management and the `helidon-bom` (see quickstart examples). I prefer the last option since it is closest to what we want our users to do.

Now that many examples share a common parent we could define `helidon.version` at a higher level pom, but no need to do that in this PR.

Reply:
Hmm; interesting.  I don't like `${helidon.version}` since it can be redefined on the Maven command line, whereas `${project.version}` cannot.  But I'm happy to use whatever is the _de facto_ standard here.

Reply:
Added `helidon.version` property.  I'm still not sure I like this.  This now gets checked in, and if we move to a new version of Helidon isn't it easy to forget to change this property?  I also added the BOM, but it doesn't seem to govern the microprofile stuff, so I ended up having to use the `${helidon.version}` value anyway.

Reply:
Yep, @romain-grecourt agrees with you. So how about this for a convention:

- Examples that are part of the reactor (have parent pom) we do explicit dependency management with project.version. Same rules apply as any module.
- Examples that are stand alone (like the quickstarts) we use dependency management with bom pom and a helidon.version property.

Apologies for the whiplash as we work out these best practices.

Reply:
Keeping `${project.version}`.